### ISSUE TITLE
[FEAT] Ignore filetypes

### ISSUE THREAD

**Issue by ls-devs** [hinell/lsp-timeout.nvim#10]
### Add option to ignore filetypes

It would be great if this plugin could ignore some filetypes like `NeoTree` `Telescope` etc..
Actually if u loose focus on a window such like `NeoTree` `Telescope` , the plugin try to restart some LSP cause its found 0 servers _(even if u just ALT-TAB quickly)_.
But it needs to restart LSPs if they stopped even if focus is regained on an ignored filetype.

**Comment by hinell** [comment]
@ls-devs Are you requesting to prevent `lsp-timeout` from starting up servers if currently focused nvim windows is in `ignored` list?

**Comment by hinell** [comment]
@ls-devs Are you requesting an option to prevent `lsp-timeout` from starting up servers if currently focused nvim windows is in `ignored` list?

**Comment by hinell** [comment]
@ls-devs Are you requesting an option to prevent `lsp-timeout` from starting up servers if currently focused nvim window is in `ignored` list?

**Comment by ls-devs** [comment]
@hinell Yes. But in certains conditions : 
- `lsp-timeout` did not shutdown the servers.
- The window focused is in `ignored` list.

Because actually, `lsp-timeout` try to restart servers if the focused window is type of `neo-tree` `Telescope` etc.. even if the LSP servers did not shutdown, because `lsp-timeout` do not find any active LSP in this window.

**Comment by hinell** [comment]
> _try to restart servers if the focused window is type of_

`lsp-timeout` is tied to a system application window, not just local `nvim-windows`.

**Comment by hinell** [comment]
I'm thinking about the following config design:

```lua
config.filepatterns.include  = {}  -- autocmd patterns for files on which LSP will be started/stopped
config.filepatterns.exclude  = {}  -- lua patterns for files which are going to be ignored by `lsp-timeout`
codfig.filetypes.ignore      = {}  -- like above, but for file types whole strings
```

Any thoughts / suggestions / objections?

**Comment by hinell** [comment]
I'm thinking about the following config design:

```lua
config.paths.include     = {}  -- autocmd patterns for files on which LSP will be started/stopped
config.paths.exclude     = {}  -- lua patterns for files which are going to be ignored by `lsp-timeout`
codfig.filetypes.ignore  = {}  -- like above, but for file types whole strings
```

Any thoughts / suggestions / objections?

**Comment by ls-devs** [comment]
@hinell I think `codfig.filetypes.ignore  = {}` should be enough for the user to do what he really wants to. 
Not needed to add so complexity for such simple thing.

**Comment by hinell** [comment]
Fixed in v1.1.0 release.

I've made a decision to avoid extending config to keep it simple.
Now, certain windows are going to be ignored. No ignore lists are used.

**Comment by hinell** [comment]
Going to address this in `v1.2.0`

**Comment by hinell** [comment]
Should to addressed in new `v1.2.0` release, checkout
